src/confighttp.cpp

```diff
@@ -522,74 +543,78 @@ namespace confighttp {
 
     std::stringstream ss;
     ss << request->content.rdbuf();
-
-    BOOST_LOG(info) << config::stream.file_apps;
     try {
       // TODO: Input Validation
-      pt::ptree fileTree;
-      pt::ptree inputTree;
-      pt::ptree outputTree;
-      pt::read_json(ss, inputTree);
-      pt::read_json(config::stream.file_apps, fileTree);
-
-      if (inputTree.get_child("prep-cmd").empty()) {
-        inputTree.erase("prep-cmd");
+      nlohmann::json output_tree;
+      nlohmann::json input_tree = nlohmann::json::parse(ss);
+      std::string file = file_handler::read_file(config::stream.file_apps.c_str());
+      BOOST_LOG(info) << file;
+      nlohmann::json file_tree = nlohmann::json::parse(file);
+
+      if (input_tree["prep-cmd"].empty()) {
+        input_tree.erase("prep-cmd");
       }
 
-      if (inputTree.get_child("detached").empty()) {
-        inputTree.erase("detached");
+      if (input_tree["detached"].empty()) {
+        input_tree.erase("detached");
       }
 
-      auto &apps_node = fileTree.get_child("apps"s);
-      int index = inputTree.get<int>("index");
+      auto &apps_node = file_tree["apps"];
+      int index = input_tree["index"].get<int>();  // this will cause exception if not a string
```

Comment:
"if not a string"?

Reply:
```suggestion
      int index = input_tree["index"].get<int>();  // this will intentionally cause exception if the provided value is the wrong type
```